src/libawkward/array/NumpyArray.cpp

```diff
@@ -602,6 +601,11 @@ namespace awkward {
     return std::vector<std::string>();
   }
 
+  int64_t NumpyArray::list_depth() const {
+    // NumpyArray is a leaf
+    return (int64_t)1;
```

Comment:
Isn't the `list_depth` of a `NumpyArray` equal to `ndim()`?

Reply:
yes, indeed.

Reply:
Removed for now - ```purist_depth``` is used in this PR